File: 2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int left = 0;
        int right = s.size()-1;
        
        while(left <= right){
            if(s[left] == '1'){
                left++;
            } 
            if(s[right] == '0'){
                right--;
            }
            if(s[left] == '0' && s[right] == '1'){
                s[left] = '1';
                s[right] = '0';
                left++;
                right--;
            }
        }
        
        //swap last '1' with last digit
        char temp = s[left-1];
        s[left-1] = s[s.size()-1];
        s[s.size()-1] = temp;
        
        return s;
    }
};

File: 2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.py
class Solution(object):
    def maximumOddBinaryNumber(self, s):
        """
        :type s: str
        :rtype: str
        """
        cnt1 = 0
        cnt0 = 0
        for i in s:
            if i == '1':
                cnt1+=1
            elif i == '0':
                cnt0+=1
                
        return ('1'*(cnt1-1)) + '0'*cnt0 + '1'